backend/editor/OT/OTClient/operation.ts

```diff
@@ -11,42 +9,50 @@ export interface Operation {
 // TODO: in the future update object operation to strictly contain CMS operation data
 export interface AtomicOperation {
   type: string;
+  opType: "insert" | "delete";
   transformAgainst: (op: AtomicOperation) => AtomicOperation[];
 }
 
 export class IntegerOperation implements AtomicOperation {
   type = "integerOperation";
   newValue: number;
-  constructor(newValue: number) {
+  opType: "insert" | "delete";
+  constructor(newValue: number, opType: "insert" | "delete") {
     this.newValue = newValue;
+    this.opType = opType;
   }
-
   transformAgainst = (op: AtomicOperation): AtomicOperation[] => [this, op];
 }
 
 export class BooleanOperation implements AtomicOperation {
   type = "booleanOperation";
   newValue: boolean;
-  constructor(newValue: boolean) {
+  opType: "insert" | "delete";
+  constructor(newValue: boolean, opType: "insert" | "delete") {
```

Comment:
There is a lot of unneeded duplication here, abstract this out into its own type:
```ts
type OperationType = "insert" | "delete";

export class BooleanOperation implements AtomicOperation {
  type = "booleanOperation";
  newValue: boolean;
  constructor(newValue: boolean) {
  opType: OperationType;
  constructor(newValue: boolean, opType: OperationType) {
       // ....
  }
}
```

Reply:
wait i thought that's what i'm supposed to do according to [this jira ticket](https://csesoc.atlassian.net/browse/CMS-307)?